tests/performance/perftest/tests.py

```diff
@@ -0,0 +1,388 @@
+        )
+        result_data.append(
+            {
+                "info": {
+                    "test_name": name,
+                },
+                "metrics": [
+                    {
+                        "name": "megabytes_per_sec",
+                        "type": "MEDIAN",
+                        "value": megabytes_per_sec,
+                        "metadata": {
+                            "improvement_direction": "up",
+                            "measurement_unit": "megabytes_per_second",
+                        },
+                    },
+                ],
+            }
+        )
+
+    def before(self):
+        pass
+
+    def do_task(self):
+        raise NotImplementedError
+
+    def after(self):
+        pass
+
+    def percentile(self, percentile):
+        if hasattr(self, "results"):
+            sorted_results = sorted(self.results)
+            percentile_index = int(len(sorted_results) * percentile / 100) - 1
+            return sorted_results[percentile_index]
+        self.fail("Test execution failed")
+        return None
+
+    def runTest(self):
+        results = []
+        start = time.monotonic()
+        i = 0
+        while True:
+            i += 1
+            self.before()
+            with Timer() as timer:
+                self.do_task()
+            self.after()
+            results.append(timer.interval)
+            duration = time.monotonic() - start
+            if duration > MIN_ITERATION_TIME and i >= NUM_ITERATIONS:
+                break
+            if duration > MAX_ITERATION_TIME:
+                with warnings.catch_warnings():
+                    warnings.simplefilter("default")
+                    warnings.warn(
+                        f"{self.__class__.__name__} timed out after {MAX_ITERATION_TIME}s, "
+                        f"completed {i}/{NUM_ITERATIONS} iterations.",
+                        stacklevel=2,
+                    )
+
+                break
+
+        self.results = results
+
+
+class SmallFlatDocTest(PerformanceTest):
+    dataset = "small_doc.json"
+
+    def setUp(self):
+        super().setUp()
+        with open(Path(TEST_PATH) / Path("flat_models") / self.dataset) as data:  # noqa: PTH123
+            self.document = json.load(data)
+
+        self.data_size = len(encode(self.document)) * NUM_DOCS
+        self.documents = [self.document.copy() for _ in range(NUM_DOCS)]
+
+
+class TestSmallFlatDocCreation(SmallFlatDocTest, TestCase):
+    def do_task(self):
+        for doc in self.documents:
+            model = SmallFlatModel(**doc)
+            model.save()
+
+    def after(self):
+        SmallFlatModel.objects.all().delete()
+
+
+class TestSmallFlatDocUpdate(SmallFlatDocTest, TestCase):
+    def setUp(self):
+        super().setUp()
+        for doc in self.documents:
+            model = SmallFlatModel(**doc)
+            model.save()
+        self.models = list(SmallFlatModel.objects.all())
```

Comment:
self.models.append(model) should suffice, especially if you use `Models.objects.create`. Removes the need for database call.